lib/Rfc1035StubResolver.php

```diff
@@ -88,9 +94,22 @@ public function resolve(string $name, int $typeRestriction = null): Promise
         }
 
         return call(function () use ($name, $typeRestriction) {
-            if (!$this->config) {
+            if ($this->configStatus === self::CONFIG_NOT_LOADED) {
                 yield $this->reloadConfig();
             }
+            if ($this->configStatus === self::CONFIG_FAILED) {
+                if (!\in_array($typeRestriction, [Record::A, null])) {
+                    throw new DnsException("All query attempts failed for {$name}: could not load DNS config, and fallback cannot load non-A queries");
+                }
+                $result = \gethostbynamel($name);
+                if ($result === false) {
+                    throw new DnsException("All query attempts failed for {$name}: could not load DNS config, and no records were returned by fallback");
```

Comment:
Should probably say `Could not resolve '{$name}' using blocking fallback mode, because either the query failed or no records were returned.`